src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/UpdateTribunalDecisionErrorRule31MidEvent.java

```diff
@@ -0,0 +1,94 @@
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.Document;
+import uk.gov.hmcts.reform.iacaseapi.domain.entities.ccd.field.YesOrNo;
+import uk.gov.hmcts.reform.iacaseapi.domain.handlers.PreSubmitCallbackHandler;
+
+@Component
+public class UpdateTribunalDecisionErrorRule31MidEvent implements PreSubmitCallbackHandler<AsylumCase> {
+
+    @Override
+    public boolean canHandle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        requireNonNull(callbackStage, "callbackStage must not be null");
+        requireNonNull(callback, "callback must not be null");
+
+        return callbackStage == PreSubmitCallbackStage.MID_EVENT
+               && callback.getEvent() == Event.UPDATE_TRIBUNAL_DECISION
+               && callback.getPageId().equals("decisionAndReasonsDocumentUploadPage");
+    }
+
+    @Override
+    public PreSubmitCallbackResponse<AsylumCase> handle(PreSubmitCallbackStage callbackStage, Callback<AsylumCase> callback) {
+        if (!canHandle(callbackStage, callback)) {
+            throw new IllegalStateException("Cannot handle callback");
+        }
+
+        final AsylumCase asylumCase = callback.getCaseDetails().getCaseData();
+        PreSubmitCallbackResponse<AsylumCase> response = new PreSubmitCallbackResponse<>(asylumCase);
+
+        final Optional<Document> decisionAndReasonsDoc = asylumCase.read(DECISION_AND_REASON_DOCS_UPLOAD, Document.class);
+
+        YesOrNo isDecisionAndReasonDocumentBeingUpdated = asylumCase.read(AsylumCaseFieldDefinition.UPDATE_TRIBUNAL_DECISION_AND_REASONS_FINAL_CHECK, YesOrNo.class)
+            .orElse(NO);
+        Optional<DynamicList> optionalTypesOfUpdateTribunalDecision = asylumCase.read(AsylumCaseFieldDefinition.TYPES_OF_UPDATE_TRIBUNAL_DECISION, DynamicList.class);
+
+        if (isDecisionAndReasonDocumentBeingUpdated == YES) {
+
+            if (decisionAndReasonsDoc.isEmpty()) {
+                response.addError("Amended Decision And Reasons Document must be present");
+                return response;
+            }
+
+            if (!decisionAndReasonsDoc.get().getDocumentFilename().endsWith(".pdf")) {
+                response.addError("The Decision and reasons document must be a PDF file");
+                return response;
+            }
+
+            Document previousDecisionAndReasonsDoc = decisionAndReasonsDoc.get();
+
+            Document amendedDecisionAndReasonsDoc = new Document(previousDecisionAndReasonsDoc.getDocumentUrl(), previousDecisionAndReasonsDoc.getDocumentBinaryUrl(),getDecisionAndReasonsFilename(asylumCase));
+
+            asylumCase.write(DECISION_AND_REASON_DOCS_UPLOAD, amendedDecisionAndReasonsDoc);
+            return new PreSubmitCallbackResponse<>(asylumCase);
+
+        } else if (isDecisionAndReasonDocumentBeingUpdated == NO &&
+            optionalTypesOfUpdateTribunalDecision.isPresent() &&
+            "No".equals(optionalTypesOfUpdateTribunalDecision.get().getValue().getLabel())) {
```

Comment:
For visibility, can we flip the equals on this?
(optionalTypesOfUpdateTribunalDecision.get().getValue().getLabel()..equals("No")